evalml/automl/engine/engine_base.py

```diff
@@ -74,8 +74,12 @@ def train_and_score_pipeline(pipeline, automl, full_X_train, full_y_train):
                 logger.debug(f"Skipping fold {i} because CV for stacked ensembles is not supported.")
                 break
             logger.debug(f"\t\tTraining and scoring on fold {i}")
-            X_train, X_valid = full_X_train.iloc[train], full_X_train.iloc[valid]
-            y_train, y_valid = full_y_train.iloc[train], full_y_train.iloc[valid]
+            if getattr(automl.data_splitter, "transform", None):
+                X_train, y_train = train[0], train[1]
+                X_valid, y_valid = valid[0], valid[1]
+            else:
+                X_train, X_valid = full_X_train.iloc[train], full_X_train.iloc[valid]
+                y_train, y_valid = full_y_train.iloc[train], full_y_train.iloc[valid]
```

Comment:
Oof, yeah. I would categorize this as tech debt. I understand this is necessary because the oversampling methods generate new rows, whereas the existing sklearn-style splitters we support do not.

I'll leave this question here for consideration although I don't have an answer right away: is there anything we can do to our existing data splitter API to make this easier?